uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -140,6 +140,8 @@ define([
         this._reset();
         // reinitialize opening balance
         setInitialAbsenceTypes.call(this);
+
+        this._calculateOpeningAndClosingBalance();
```

Comment:
As I mentioned in my earlier comment, `setInitialAbsenceTypes` (previously named as `initAbsenceType`) should be removed from `changeInNoOfDays`. Because in the UI Single Day or Multiple Day is clicked the absence type should not be set again, only the balance should be calculated as the No of days changed.